lib/bloc/chainswap/chainswap_bloc.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_breez_liquid/flutter_breez_liquid.dart';
+import 'package:l_breez/bloc/account/breez_sdk_liquid.dart';
+import 'package:l_breez/bloc/account/payment_error.dart';
+import 'package:l_breez/bloc/chainswap/chainswap_state.dart';
+
+class ChainSwapBloc extends Cubit<ChainSwapState> {
+  final BreezSDKLiquid _liquidSdk;
+
+  ChainSwapBloc(this._liquidSdk) : super(ChainSwapState.initial());
+
+  Future<OnchainPaymentLimitsResponse> fetchOnchainLimits() async {
+    return await _liquidSdk.instance!.fetchOnchainLimits();
+  }
+
+  Future<PreparePayOnchainResponse> preparePayOnchain({
+    required PreparePayOnchainRequest req,
+  }) async {
+    return await _liquidSdk.instance!.preparePayOnchain(req: req);
+  }
+
+  Future<SendPaymentResponse> payOnchain({
+    required PayOnchainRequest req,
+  }) async {
+    return await _liquidSdk.instance!.payOnchain(req: req);
+  }
+
+  Future<PrepareReceiveOnchainResponse> prepareReceiveOnchain({
+    required PrepareReceiveOnchainRequest req,
+  }) async {
+    return await _liquidSdk.instance!.prepareReceiveOnchain(req: req);
+  }
+
+  Future<ReceiveOnchainResponse> receiveOnchain({
+    required PrepareReceiveOnchainResponse req,
+  }) async {
+    return await _liquidSdk.instance!.receiveOnchain(req: req);
+  }
+
+  Future<RefundResponse> refund({
+    required RefundRequest req,
+  }) async {
+    return await _liquidSdk.instance!.refund(req: req);
+  }
+
+  Future<void> rescanOnchainSwaps() async {
+    return await _liquidSdk.instance!.rescanOnchainSwaps();
+  }
+
+  void validateSwap(
+    BigInt amount,
+    bool outgoing,
+    OnchainPaymentLimitsResponse onchainLimits,
+  ) {
+    var limits = outgoing ? onchainLimits.send : onchainLimits.receive;
+    if ((limits.maxSat != BigInt.zero && amount > limits.maxZeroConfSat) || amount > limits.maxZeroConfSat) {
```

Comment:
You can check `amount > maxSat`, the `maxZeroConfSat` just informs at what amount is the lockup confirmation needed (as apposed to seen in the mempool)